app/src/main/java/com/example/sharingang/MainActivity.kt

```diff
@@ -88,22 +89,25 @@ class MainActivity : AppCompatActivity(R.layout.activity_main) {
         Firebase.dynamicLinks
             .getDynamicLink(intent)
             .addOnSuccessListener(this) { pendingDynamicLinkData ->
-                val deepLink = pendingDynamicLinkData?.link
-                val id = deepLink?.getQueryParameter("id")
-                if (deepLink?.path == "/item" && id != null) {
-                    lifecycleScope.launch(Dispatchers.IO) {
-                        val item = itemRepository.get(id)
-                        if (item != null) {
-                            runOnUiThread {
-                                getNavController().navigate(
-                                    ItemsListFragmentDirections
-                                        .actionItemsListFragmentToDetailedItemFragment(item)
-                                )
-                            }
-                        }
+                pendingDynamicLinkData?.link?.let { openDeepLink(it) }
+            }
+    }
+
+    fun openDeepLink(deepLink: Uri) {
+        val id = deepLink.getQueryParameter("id")
+        if (deepLink.path == "/item" && id != null) {
+            lifecycleScope.launch(Dispatchers.IO) {
+                val item = itemRepository.get(id)
+                if (item != null) {
+                    runOnUiThread {
+                        getNavController().navigate(
+                            ItemsListFragmentDirections
+                                .actionItemsListFragmentToDetailedItemFragment(item)
+                        )
```

Comment:
I'm surprised CodeClimate didn't complain about that 😆

Reply:
I think it's almost at the limit haha